Detectors/EMCAL/simulation/src/LZEROElectronics.cxx

```diff
@@ -0,0 +1,241 @@
+void LZEROElectronics::init()
+{
+  mSimParam = &(o2::emcal::SimParam::Instance());
+  mRandomGenerator = new TRandom3(std::chrono::high_resolution_clock::now().time_since_epoch().count());
+  mSimulateNoiseDigits = mSimParam->doSimulateNoiseDigits();
+  mTriggerMap = new TriggerMappingV2(mGeometry);
+  setThreshold(mSimParam->getThresholdLZERO());
+  // setThreshold(132.);
+}
+//________________________________________________________
+void LZEROElectronics::clear()
+{
+}
+//________________________________________________________
+void LZEROElectronics::updatePatchesADC(Patches& p)
+{
+  p.updateADC();
+}
+//_______________________________________________________________________
+void LZEROElectronics::addNoiseDigits(Digit& d1)
+{
+  double amplitude = d1.getAmplitude();
+  double sigma = mSimParam->getPinNoiseTRU();
+  // double sigma = 0.04;
+
+  uint16_t noise = std::floor(std::abs(mRandomGenerator->Gaus(0, sigma) / constants::EMCAL_TRU_ADCENERGY)); // ADC
+
+  Digit d(d1.getTower(), 0., noise, d1.getTimeStamp());
+
+  d1 += d;
+}
+//_______________________________________________________________________
+void LZEROElectronics::fill(const std::deque<o2::emcal::DigitTimebinTRU>& digitlist, const o2::InteractionRecord record, std::vector<Patches>& patchesFromAllTRUs)
+{
+  int counterDigitTimeBin = 0;
+  int sizemDigitMap = -999;
+
+  for (auto& digitsTimeBin : digitlist) {
+    // Inside the DigitTimebinTRU
+    // Fill the LZEROElectronics with the new ADC value
+    // At the end of the loop run the peak finder
+    // Ship to LONEElectronics in case a peak is found
+    // Entire logic limited to timebin by timebin -> effectively implementing time scan
+
+    counterDigitTimeBin++;
+
+    int counterhelp = 0;
+    for (auto& [fastor, digitsList] : *digitsTimeBin.mDigitMap) {
+      // Digit loop
+      // The peak finding algorithm is run after getting out of the loop!
+
+      auto whichTRU2 = std::get<0>(mTriggerMap->getTRUFromAbsFastORIndex(fastor));
+      auto whichFastOrTRU2 = std::get<1>(mTriggerMap->getTRUFromAbsFastORIndex(fastor));
+
+      if (digitsList.size() == 0) {
+        continue;
+      }
+      digitsList.sort();
+
+      int digIndex = 0;
+      Digit summedDigit;
+      bool first = true;
+      for (auto& ld : digitsList) {
+        if (first) {
+          summedDigit = ld;
+          first = false;
+        } else {
+          // summedDigit += ld;
+
+          // safety device in case same fastOr
+          // but different towers, i.e. remember
+          // that the += operator fails WITHOUT
+          // feedback if that were to happen
+          Digit digitToSum(summedDigit.getTower(), ld.getAmplitude(), summedDigit.getTimeStamp());
+          summedDigit += digitToSum;
+        }
+      }
+
+      sizemDigitMap = (*digitsTimeBin.mDigitMap).size();
+      if (mSimulateNoiseDigits) {
+        addNoiseDigits(summedDigit);
+      }
+
+      if (summedDigit.getAmplitude() < 0) {
+        continue;
+      }
```

Comment:
Can this ever happen? (Probably a remanent of the run2 code where some ADCs were initialised with -1)

Reply:
In principle no, it is as you say, a remnant of run2 code. Please let me know if you would like it removed or if it should be kept.

Reply:
Ok, I am removing it.

Reply:
Done